src/mysky.ts

```diff
@@ -345,31 +685,67 @@ export class MySky {
         return;
       }
 
-      // Parse the seed.
-      const seed = new Uint8Array(JSON.parse(newValue));
-
-      // Launch the new permissions provider.
-      this.permissionsProvider = launchPermissionsProvider(seed);
+      try {
+        // Parse the seed.
+        const seed = new Uint8Array(JSON.parse(newValue));
 
-      // If the email is found, then set up auto-login on Main MySky.
-      const email = localStorage.getItem(EMAIL_STORAGE_KEY);
-      if (email) {
-        // Clear the stored email.
+        // Connect to siasky.net first.
         //
-        // The email can be cleared here because `localStorage` is only used to
-        // marshal the email from MySky UI over to the invisible MySky iframe.
-        // We don't clear the seed because we need it in storage so that users
-        // are automatically logged-in, when possible. But for the email, it
-        // should be stored on MySky, as the local storage can get cleared,
-        // users can move across browsers etc.
-        localStorage.removeItem(EMAIL_STORAGE_KEY);
-
-        // TODO: Set up auto-login.
+        // NOTE: Don't use the stored preferred portal here because we are just
+        // logging into a new account and need to get the user settings for the
+        // first time. Always use siasky.net.
+        this.client = getLoginClient(seed, null);
+
+        // Login to portal.
+        {
+          // Get the preferred portal and switch to it.
+          const { portal: preferredPortal, email } = await this.getUserSettings(seed);
+          let storedEmail = email;
+
+          // Set the portal
+          this.setPortal(preferredPortal);
+
+          // The email wasn't in the user settings but the user might have just
+          // signed up with it -- check local storage. We don't need to do this if
+          // the email was already found.
+          // TODO: Add dedicated flow(s) for changing the email after it's set.
+          let isEmailProvided = false;
+          if (!storedEmail) {
+            storedEmail = checkStoredEmail();
+            isEmailProvided = storedEmail !== null;
+          }
+
+          if (storedEmail) {
```

Comment:
this saves you a bunch of lines and a boolean, in any case I would rename `isEmailProvided` to something that explains why it will be used, or update the user settings inside of the `if (!storedEmail) {`, both would read easier

Reply:
Renamed the variable and added an explanatory comment.